aspnetcore/mvc/controllers/routing.md

```diff
@@ -167,570 +235,656 @@ public class ProductsController : Controller
-> REST APIs should use attribute routing to model the app's functionality as a set of resources where operations are represented by HTTP verbs. This means that many operations (for example, GET, POST) on the same logical resource will use the same URL. Attribute routing provides a level of control that's needed to carefully design an API's public endpoint layout.
+* [[HttpGet]](xref:Microsoft.AspNetCore.Mvc.HttpGetAttribute)
+* [[HttpPost]](xref:Microsoft.AspNetCore.Mvc.HttpPostAttribute)
+* [[HttpPut]](xref:Microsoft.AspNetCore.Mvc.HttpPutAttribute)
+* [[HttpDelete]](xref:Microsoft.AspNetCore.Mvc.HttpDeleteAttribute)
+* [[HttpHead]](xref:Microsoft.AspNetCore.Mvc.HttpHeadAttribute)
+* [[HttpPatch]](xref:Microsoft.AspNetCore.Mvc.HttpPatchAttribute)
 
-Since an attribute route applies to a specific action, it's easy to make parameters required as part of the route template definition. In this example, `id` is required as part of the URL path.
+<a name="rt"></a>
 
-```csharp
-public class ProductsApiController : Controller
-{
-   [HttpGet("/products/{id}", Name = "Products_List")]
-   public IActionResult GetProduct(int id) { ... }
-}
-```
+### Route templates
 
-The `ProductsApi.GetProduct(int)` action will be executed for a URL path like `/products/3` but not for a URL path like `/products`. See [Routing](xref:fundamentals/routing) for a full description of route templates and related options.
+ASP.NET Core has the following route templates:
 
-## Route Name
+* All the [HTTP verb templates](#verb) are route templates.
+* [[Route]](xref:Microsoft.AspNetCore.Mvc.RouteAttribute)
 
-The following code  defines a *route name* of `Products_List`:
+<a name="arx"></a>
 
-```csharp
-public class ProductsApiController : Controller
-{
-   [HttpGet("/products/{id}", Name = "Products_List")]
-   public IActionResult GetProduct(int id) { ... }
-}
-```
+## Attribute routing with Http verb attributes
 
-Route names can be used to generate a URL based on a specific route. Route names have no impact on the URL matching behavior of routing and are only used for URL generation. Route names must be unique application-wide.
+<!-- I had to add this sample code/explanation, even though you can get it elsewhere in the doc. But the doc doesn't put it all together in one place. I need an H2 to link to this information because we get so many issues on https://docs.microsoft.com/en-us/aspnet/core/tutorials/first-web-api where they don't understand constraints and how the route is determined. I may move this section to an include and add it in the Web API tutorial - Or I could move it there and not include it here.
+For example, https://github.com/aspnet/AspNetCore.Docs/issues/16901
```

Comment:
I'm not totally sure what you're trying to say - what's the point of confusion? The concatenation of templates? What would help alleviate it? Better conceptual explanation? Better examples?

I'm fine with what you added here 👍